ui/src/app/pages/admin/admin-institution-edit.tsx

```diff
@@ -0,0 +1,422 @@
+      invalidEmailDomainsMsg: '',
+      showOtherInstitutionTextBox: false,
+      showBackButtonWarning: false,
+      showApiError: false
+    };
+  }
+
+  async componentDidMount() {
+    // If institution short Name is passed in the URL get the institution details
+    if (this.props.urlParams.institutionId) {
+      institutionToEdit = await institutionApi().getInstitution(this.props.urlParams.institutionId);
+      title = institutionToEdit.displayName;
+      this.setState({
+        institutionMode: InstitutionMode.EDIT,
+        institution: institutionToEdit,
+        showOtherInstitutionTextBox: institutionToEdit.organizationTypeEnum === OrganizationType.OTHER
+      });
+    } else {
+      title = 'Add new Institution';
+      this.setState({institutionMode: InstitutionMode.ADD});
+    }
+
+  }
+
+  get institutionTypeOptions() {
+    const options = [
+      {label: 'Industry', value: OrganizationType.INDUSTRY},
+      {label: 'Academic Research Institution', value: OrganizationType.ACADEMICRESEARCHINSTITUTION},
+      {label: 'Educational Institution', value: OrganizationType.EDUCATIONALINSTITUTION},
+      {label: 'Health Center/ Non-Profit', value: OrganizationType.HEALTHCENTERNONPROFIT},
+      {label: 'Other', value: OrganizationType.OTHER}
+    ];
+    return options;
+  }
+
+  get institutionAgreementTypeOptions() {
+    const options = [
+      {label: 'Master', value: DuaType.MASTER},
+      {label: 'Individual', value: DuaType.RESTRICTED}
+    ];
+    return options;
+  }
+
+  // Filter out empty line or empty email addresses like <email1>,,<email2>
+  // Confirm each email is a valid email using validate.js
+  validateEmailAddresses() {
+    const invalidEmailAddress = [];
+    const {emailAddresses} = this.state.institution;
+    const emailAddressWithNO = emailAddresses.filter(
+      emailAddress => {
+        return emailAddress !== '' || !!emailAddress;
+      });
+
+    this.setState(fp.set(['institution', 'emailAddresses'], emailAddressWithNO));
+    this.state.institution.emailAddresses.map(emailAddress => {
+      const errors = validate({
+        emailAddress
+      }, {
+        emailAddress: {email: true}
+      });
+      if (errors && errors.emailAddress && errors.emailAddress.length > 0) {
+        invalidEmailAddress.push(emailAddress);
+      }
+    });
+    this.setState({invalidEmailAddress: invalidEmailAddress.length > 0});
+    if (invalidEmailAddress.length > 0) {
+      const errMessage = 'Following Email Addresses are not valid : ' + invalidEmailAddress.join(' , ');
+      this.setState({invalidEmailAddressMsg: errMessage});
+    }
+    return;
+  }
+
+  // Filter out empty line or empty email addresses like <email1>,,<email2>
+  // Confirm each email domain matches with regex
+  validateEmailDomains() {
+    const invalidEmailDomain = [];
+    const {emailDomains} = this.state.institution;
+    this.state.institution.emailDomains =  emailDomains.filter(emailDomain => emailDomain);
```

Comment:
this looks like another case of unsafe state setting.  also, what is it intended to do?

Reply:
remove empty emailDomains "If you want to keep only truthy values you may use .filter(e => e)" however i might use trim as well just to avoid strings like '    '